src/plugins/intel_gpu/src/kernel_selector/cl_kernels/include/batch_headers/fetch_data.cl

```diff
@@ -715,25 +749,30 @@ inline uint get_bs_fs_zyx_bsv_fsv_index(uint b, uint f,  uint z, uint y, uint x,
     const uint fs_pitch = z_pitch * (z_pad_before + z_size + z_pad_after);
     const uint total_f_size = f_pad_before + f_size + f_pad_after;
     const uint bs_pitch = fs_pitch * ((total_f_size + f_alignment - 1) / f_alignment);
+    const uint bs_pad_before = b_pad_before / b_alignment;
+    const uint bsv_pad_before = b_pad_before % b_alignment;
 
-    const uint output_offset = bs * bs_pitch +
+    const uint output_offset = (bs_pad_before + bs) * bs_pitch +
                                fs * fs_pitch +
                                (z_pad_before + z) * z_pitch +
                                (y_pad_before + y) * y_pitch +
                                (x_pad_before + x) * x_pitch +
-                               bsv * bsv_pitch
+                               (bsv_pad_before + bsv) * bsv_pitch
```

Comment:
`bsd_pad_before + bsv` can be higher then 32. I don't think this is a correct implementation.

Reply:
You are right. It is supposed to be added before calculating bsv.
    const uint bs = (b + b_pad_before) / b_alignment;
    const uint bsv = (b + b_pad_before) % b_alignment;

Reply:
A correct calc is even simpler. it is fixed.